src/litserve/server.py

```diff
@@ -500,24 +500,40 @@ async def win_data_streamer(self, read, write):
 
     async def data_streamer(self, read: Connection, write: Connection):
         data_available = asyncio.Event()
-        while True:
-            # Calling poll blocks the event loop, so keep the timeout low
-            if not read.poll(0.001):
-                asyncio.get_event_loop().add_reader(read.fileno(), data_available.set)
+        queue = asyncio.Queue()
+        loop = asyncio.get_event_loop()
+
+        def reader():
+            try:
+                while read.poll(0.001):  # Check if there's data available to read
+                    response, status = read.recv()
+                    queue.put_nowait((response, status))
+                    data_available.set()
+            except Exception as e:
+                logger.error(f"Exception in reader: {e}")
+
+        loop.add_reader(read.fileno(), reader)
+
+        try:
+            while True:
                 await data_available.wait()
                 data_available.clear()
```

Comment:
couldn't there be a scenario where
- `reader` is looping calls `queue.put_nowait` and `data_available.set()`
- the loop further down unblocks, clears `data_available`
- it consumes all items in the queue, while `reader` is still pushing and setting `data_available` afterwards
- the queue now is empty but `data_available` is set
- the loop loops one last time, `data_available` passes, but there are no more items on the queue

ah ok, in that case `queue` is empty and no harm is done. Just wanted to check :-)

Reply:
yes, this is the expected flow.